src/cache.zig

```diff
@@ -0,0 +1,177 @@
+// Spec here https://github.com/roc-lang/roc/issues/7517
+const builtin = @import("builtin");
+const std = @import("std");
+const testing = std.testing;
+const Allocator = std.mem.Allocator;
+
+pub const CompilerCacheFile = struct {
+    version_name: []const u8,
+    binary: []const u8,
+};
+
+pub const RocCacheFileTag = enum {
+    compiler,
+    build,
+    packages_src,
+    packages_build,
+};
+
+pub const RocCacheFile = union(RocCacheFileTag) {
```

Comment:
Two naming things:
- Roc is implied throughout this whole compiler, so no need to prefix names with it except in cases of ambiguity
- Zig prefers to nest related "has-a" data like the above `CompilerCacheFile`: https://ziggit.dev/t/nesting-struct-types-or-not/6750

An example of how I might nest it would be:

```zig
pub const CacheFile = union(enum) {
    compiler: Compiler,
    build,
    packages_src,
    packages_build,

    pub const Compiler = struct {
        version_name: []const u8,
        binary: []const u8,
    };
};
```